source/materials/OpticalMaterialProperties.cc

```diff
@@ -20,16 +23,19 @@ G4MaterialPropertiesTable* OpticalMaterialProperties::Vacuum()
 {
   G4MaterialPropertiesTable* mpt = new G4MaterialPropertiesTable();
 
-  const G4int entries = 2;
-  G4double energy[entries] = {0.01*eV, 100.*eV};
+  std::vector<G4double> energy = {optPhotMinE_, optPhotMaxE_};
 
   // REFRACTIVE INDEX
-  G4double rindex[entries] = {1., 1.};
-  mpt->AddProperty("RINDEX", energy, rindex, entries);
+  std::vector<G4double> ri_index = {1., 1.};
+
+  assert(energy.size() == ri_index.size());
```

Comment:
In this case, it won't fail, but if I had made a mistake in the size of the vectors, the compiler wouldn't complain, and Geant4 would not, either (it would just cut the longer vector to the same size of the other one) leading to an incorrect behaviour. Instead, with this assert, the simulation crashes on run time with an error message that points one here. It's a test also for future modifications of the code.